apps/verticalhome/js/app.js

```diff
@@ -88,6 +107,19 @@
       }
     },
 
+    isInPinnedApps : function(result) {
+      var pinnedItems = app.pinnedAppsManager.items;
+      var resManifest = result.app.manifestURL;
+
+      for (var i=0; i<pinnedItems.length; i++){
+        if(pinnedItems[i].entry!=null &&
+          (resManifest == pinnedItems[i].entry.manifestURL)){
+          return true;
+        }
+      }
```

Comment:
Is it works good when we try to pin applications from communications folder which has similar manifestURL?
May be it is necessary to add comparison entry points if it is exists and if manifestURL is equal.